contracts/contracts/strategies/plume/RoosterAMOStrategy.sol

```diff
@@ -0,0 +1,1226 @@
+        }
+    }
+
+    /**
+     * @dev Collect stkAave, convert it to AAVE send to Vault.
+     */
+    function collectRewardTokens()
+        external
+        override
+        onlyHarvester
+        nonReentrant
+    {
+        // Do nothing if there's no position minted
+        if (tokenId > 0) {
+            uint32[] memory binIds = new uint32[](1);
+            binIds[0] = 1;
+
+            uint256 lastEpoch = votingDistributor.lastEpoch();
+
+            poolDistributor.claimLp(
+                address(this),
+                tokenId,
+                mPool,
+                binIds,
+                lastEpoch
+            );
+        }
+
+        // Run the internal inherited function
+        _collectRewardTokens();
+    }
+
+    /***************************************
+            Balances and Fees
+    ****************************************/
+
+    /**
+     * @dev Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        require(_asset == WETH, "Only WETH supported");
+
+        // because of PoolLens inaccuracy there is usually some dust WETH left on the contract
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        // just paranoia check, in case there is OETHb in the strategy that for some reason hasn't
+        // been burned yet. This should always be 0.
+        uint256 _oethpBalance = IERC20(OETHp).balanceOf(address(this));
+        return underlyingAssets + _wethBalance + _oethpBalance;
+    }
+
+    /**
+     * @notice Strategy reserves (which consist only of WETH in case of Rooster - Plume pool)
+     * when the tick price is closest to parity - assuring the lowest amount of tokens
+     * returned for the current position liquidity.
+     */
+    function _balanceInPosition() internal view returns (uint256 _wethBalance) {
+        (, , uint256 liquidity) = _getPositionInformation();
+
+        uint256 _oethbBalance;
+
+        (
+            _wethBalance,
+            _oethbBalance
+        ) = _reservesInTickForGivenPriceAndLiquidity(
+            sqrtPriceTickLower,
+            sqrtPriceTickHigher,
+            sqrtPriceAtParity,
+            liquidity
+        );
+
+        require(_oethbBalance == 0, "Non zero oethbBalance");
+    }
+
+    /**
+     * @notice Tick dominance denominated in 1e18
+     *
+     */
+    function tickDominance() public view returns (uint256 _tickDominance) {
+        IMaverickV2Pool.TickState memory tickState = mPool.getTick(TICK_NUMBER);
+
+        uint256 wethReserve = tickState.reserveA;
+        uint256 oethpReserve = tickState.reserveB;
+
+        (
+            uint256 _amountWeth,
+            uint256 _amountOethp,
+
+        ) = _getPositionInformation();
+
+        if ((wethReserve + oethpReserve) > 0) {
```

Comment:
⚪ Clearer code flow if we early return for the zero case.

Reply:
Agreed: https://github.com/OriginProtocol/origin-dollar/pull/2499/commits/58f849f1e0338d821a375a2fa16e102ad8626a6c